web/src/app/alerts/AlertsList.js

```diff
@@ -0,0 +1,369 @@
+import {
+  ArrowUpward as EscalateIcon,
+  Check as AcknowledgeIcon,
+  Close as CloseIcon,
+  Info as InfoIcon,
+} from '@material-ui/icons'
+import { useSelector } from 'react-redux'
+
+import AlertsListFilter from './components/AlertsListFilter'
+import AlertsListControls from './components/AlertsListControls'
+import CreateAlertFab from './CreateAlertFab'
+import UpdateAlertsSnackbar from './components/UpdateAlertsSnackbar'
+
+import { formatTimeSince } from '../util/timeFormat'
+import { urlParamSelector } from '../selectors'
+import { useMutation, useQuery } from '@apollo/react-hooks'
+import QueryList from '../lists/QueryList'
+import useWidth from '../util/useWidth'
+
+export const alertsListQuery = gql`
+  query alertsList($input: AlertSearchOptions) {
+    alerts(input: $input) {
+      nodes {
+        id
+        alertID
+        status
+        summary
+        details
+        createdAt
+        serviceID
+        service {
+          name
+        }
+      }
+
+      pageInfo {
+        endCursor
+        hasNextPage
+      }
+    }
+  }
+`
+
+const updateMutation = gql`
+  mutation UpdateAlertsMutation($input: UpdateAlertsInput!) {
+    updateAlerts(input: $input) {
+      alertID
+      id
+    }
+  }
+`
+
+const escalateMutation = gql`
+  mutation EscalateAlertsMutation($input: [Int!]) {
+    escalateAlerts(input: $input) {
+      alertID
+      id
+    }
+  }
+`
+
+const useStyles = makeStyles(theme => ({
+  snackbar: {
+    backgroundColor: theme.palette.primary['500'],
+    height: '6.75em',
+    width: '20em', // only triggers on desktop, 100% on mobile devices
+  },
+  snackbarIcon: {
+    fontSize: 20,
+    opacity: 0.9,
+    marginRight: theme.spacing(1),
+  },
+  snackbarMessage: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+}))
+
+function getStatusFilter(s) {
+  switch (s) {
+    case 'acknowledged':
+      return ['StatusAcknowledged']
+    case 'unacknowledged':
+      return ['StatusUnacknowledged']
+    case 'closed':
+      return ['StatusClosed']
+    case 'all':
+      return ['StatusAcknowledged', 'StatusUnacknowledged', 'StatusClosed']
+    // active is the default tab
+    default:
+      return ['StatusAcknowledged', 'StatusUnacknowledged']
+  }
+}
+
+export default function AlertsList(props) {
+  const classes = useStyles()
+  const width = useWidth()
+  const isFullScreen = isWidthDown('md', width)
+
+  // state initialization
+  const [errorMessage, setErrorMessage] = useState('')
+  const [updateMessage, setUpdateMessage] = useState('')
+  const [updateComplete, setUpdateComplete] = useState(false)
+
+  // get redux url vars
+  const params = useSelector(urlParamSelector)
+  const allServices = params('allServices')
+  const filter = params('filter', 'active')
+  const isFirstLogin = params('isFirstLogin')
+
+  // always open unless clicked away from or there are favorite services present
+  const [_showNoFavoritesWarning, setShowNoFavoritesWarning] = useState(true)
```

Comment:
Would this be clearer if we just track it as "dismissed"?
```suggestion
  const [noFavDismissed, SetNoFavDismissed] = useState(false)
```